lib/Controller/EpisodeActionController.php

```diff
@@ -32,17 +34,23 @@ public function __construct(
 		$this->episodeActionRepository = $episodeActionRepository;
 		$this->userId = $UserId;
 		$this->episodeActionSaver = $episodeActionSaver;
+		$this->request = $request;
 	}
 
 	/**
 	 *
 	 * @NoAdminRequired
 	 * @NoCSRFRequired
 	 *
-	 * @return Response
+	 * @return JSONResponse
 	 */
-	public function create($data) {
-		return $this->episodeActionSaver->saveEpisodeActions($data, $this->userId);
+	public function create(): JSONResponse {
+
+		$episodeActionsArray = $this->filterEpisodesFromRequestParams($this->request->getParams());
```

Comment:
I'm not sure if this is done the right way, because the array filtering could have been done right here as well. I decided to do it this way to make the code better understandable.
Filtering out non-numeric indexes is necessary, because in the request params there is always one element `"_route":"gpoddersync.episode_action.create",` that has to be filtered out.

Reply:
Don't we get the json body as first argument in the controller action? Why do you retrieve the data from class property?

Reply:
We had `{"data": [{"podcast": [...] }, {"podcast": [...] }]}`, we only got the json body as first argument because it is stored in the field "data" (before, the EpisodeAction string was stored in this field).
With my changes we expect `[{"podcast": [...] }, {"podcast": [...] }]`. This is not only very close to the gpodder api, Bytehamster even changed the payload for EpisodeAction upload to the same as gpodder.net in AntennaPod.